python/python/tests/test_pubsub.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+import pytest
+from typing import List, Set, Dict, Tuple, Optional, Any, Awaitable
+import asyncio
+import fnmatch
+
+from tests.conftest import create_client
+
+from glide.async_commands.core import (
+    CoreCommands,
+)
+
+
+MessagesByExactChannel = Dict[str, Dict[str, int]] # channel -> {msg -> cnt}
```

Comment:
I believe you can reduce the amount of duplication if you change this to `channel -> {msg -> (cnt, boolean`}
and use the added boolean to test for shardedness.